build.gradle

```diff
@@ -9,8 +9,8 @@ plugins {
 	id "org.jetbrains.kotlinx.kover" version "0.4.2"
 }
 
-sourceCompatibility = JavaVersion.VERSION_11
-targetCompatibility = JavaVersion.VERSION_11
+sourceCompatibility = JavaVersion.VERSION_17
```

Comment:
These types of changes should go into `main` and not be in this PR. Because you have done these in separate commits anyway, you can just `cherry-pick` the commits into a new branch, make a separate PR, and merge that before this thing is ready

Reply:
let me know if you have questions about this -- it should be easy and fast once you understand what you have to do

Reply:
this was rebased from main

Reply:
If you rebase onto a branch that has this commit, it shouldn't show up in the diff here -- so we're doing something wrong here